app/stuquiz/repositories/admin_repository.py

```diff
@@ -0,0 +1,36 @@
+# @author Simone Nicol <en0mia.dev@gmail.com>
+# @created 05/07/23
+from typing import Optional
+
+from app.stuquiz.entities.admin import Admin
+from app.stuquiz.repositories.abstract_repository import AbstractRepository
+
+
+class AdminRepository(AbstractRepository):
+    def select_admin_by_id(self, admin_id: str) -> Optional[Admin]:
+        """Returns the admin with the provided id.
+        :param admin_id: Admin's ID
+        :return: Admin | None
+        """
+        query = "SELECT id, username, email, password, salt FROM admin WHERE id = %s"
+        result = self.select(query, (admin_id, ))
+        return Admin(*result[0]) if result and len(result) > 0 else None
```

Comment:
The code assumes that the returned results will always contain valid data to create an Admin instance. It's recommended to include appropriate data validation and error handling in case the retrieved data is invalid or incomplete.
It is something that can be applied elsewhere. It is non-blocking, so at your discretion you can decide whether to make a change already in the authentication part or postpone the change by opening an issue.

Reply:
As we said in private chat, this is not a problem since the queries return the parameters in order.